rcl_action/src/rcl_action/action_server.c

```diff
@@ -163,6 +165,16 @@ rcl_action_server_init(
   action_server->impl->type_hash = rosidl_get_zero_initialized_type_hash();
 
   rcl_ret_t ret = RCL_RET_OK;
+  // Resolve action name
+  char * resolved_action_name = NULL;
+  ret = rcl_node_resolve_name(node, action_name, allocator, false, false, &resolved_action_name);
+  if (RCL_RET_OK != ret) {
+    if (RCL_RET_TOPIC_NAME_INVALID == ret || RCL_RET_SERVICE_NAME_INVALID == ret) {
+      ret = RCL_RET_ACTION_NAME_INVALID;
+    }
```

Comment:
rcl_node_resolve_name() can return many types of errors, so the return values should be simplified.

Reply:
I have implemented the recommended changes.